packages/ra-ui-materialui/src/list/filter/FilterButton.spec.tsx

```diff
@@ -102,12 +85,43 @@ describe('<FilterButton />', () => {
 
             fireEvent.click(getByLabelText('ra.action.add_filter'));
 
-            const disabledFilter = getByRole('menuitem');
+            const disabledFilter = queryByText('Returned')?.closest('li');
 
             expect(disabledFilter).not.toBeNull();
-            expect(disabledFilter.getAttribute('aria-disabled')).toEqual(
+            expect(disabledFilter?.getAttribute('aria-disabled')).toEqual(
                 'true'
             );
         });
+
+        it('should remove all filters when the "Clear all filters" button is clicked', async () => {
+            render(<Basic />);
+
+            // First, check we don't have a clear filters option yet
+            await screen.findByText('Add filter');
+            fireEvent.click(screen.getByText('Add filter'));
+
+            await waitFor(() => {
```

Comment:
wrong strategy: you wait until you don't find the text. But this will be true even before the list of available filters renders..

I'd wait for the list of available filters to be displayed (e.g. by finding the firts one), than text that the 'Remove all filters' text can't be found.

Reply:
I am not sure I understand but filter list is never rendered before "Add filter" is clicked (list is contained in a popper). Could we discuss it on Monday?